OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/ReachableMethodAnalysis.scala

```diff
@@ -36,18 +35,9 @@ trait ReachableMethodAnalysis extends FPCFAnalysis with TypeConsumerAnalysis {
 
     final def analyze(declaredMethod: DeclaredMethod): PropertyComputationResult = {
         val callersEOptP = propertyStore(declaredMethod, Callers.key)
-        (callersEOptP: @unchecked) match {
-            case FinalP(NoCallers) =>
-                // nothing to do, since there is no caller
-                return NoResult;
-
-            case eps: EPS[_, _] =>
-                if (eps.ub eq NoCallers) {
-                    // we can not create a dependency here, so the analysis is not allowed to create
-                    // such a result
-                    throw new IllegalStateException("illegal immediate result for callers")
-                }
-            // the method is reachable, so we analyze it!
+
+        if (callersEOptP.isFinal && callersEOptP.ub == NoCallers) {
+            return NoResult;
```

Comment:
Shouldn't we keep the check for an illegal `NoCallers` interim state at least as an assertion? Right now it is not clearly visible that the method is reachable and should be analyzed.

Reply:
The `NoCallers` interim state was illegal since the code at the top did not know the class could use the continuation without a depending caller. Since the class can invoke and handle continuations just fine without using a depender constructed from a caller, the `NoCallers` interim state is not illegal anymore.

Reply:
But it's an illegal state in general, isn't it? I don't think call graph analyses are supposed to ever create such an interim state.

Reply:
If it is an illegal state in general, then it should be asserted by the call graph generating framework and not the framework for analysis that use the call graph (even if they at least partially overlap). This class seems like the wrong area of code to make such an assertion (e.g. it would only be triggered when using such an analysis even though the result should be illegal regardless of using this class or not).